src/backend/storage/ipc/ipci.c

```diff
@@ -128,6 +130,7 @@ CreateSharedMemoryAndSemaphores(void)
 		size = add_size(size, ProcGlobalShmemSize());
 		size = add_size(size, XLOGShmemSize());
 		size = add_size(size, CLOGShmemSize());
+		size = add_size(size, CSNLogShmemSize());
```

Comment:
You missed calling `size = add_size(size, CSNSnapshotShmemSize());` and `CSNSnapshotShmemInit()` in this function, which caused segfault on startup. Maybe they were removed during code merge?

Reply:
Yes, sorry this was lost while during the merge.